android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorageTest.kt

```diff
@@ -698,11 +772,13 @@ internal fun geckoContentPermission(
     type: Int,
     value: Int = VALUE_PROMPT,
     thirdPartyOrigin: String = "mozilla.dev",
+    privateMode: Boolean = false,
 ): ContentPermission {
-    val prompt: ContentPermission = mock()
-    ReflectionUtils.setField(prompt, "uri", uri)
-    ReflectionUtils.setField(prompt, "thirdPartyOrigin", thirdPartyOrigin)
-    ReflectionUtils.setField(prompt, "permission", type)
-    ReflectionUtils.setField(prompt, "value", value)
-    return prompt
+    val permission: ContentPermission = mock()
+    ReflectionUtils.setField(permission, "uri", uri)
+    ReflectionUtils.setField(permission, "thirdPartyOrigin", thirdPartyOrigin)
+    ReflectionUtils.setField(permission, "permission", type)
+    ReflectionUtils.setField(permission, "value", value)
+    ReflectionUtils.setField(permission, "privateMode", privateMode)
+    return permission
 }
```

Comment:
We don't have any tests where we call updateGeckoPermissionIfNeeded with a private flag?

Reply:
Adding it!